crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -1933,7 +1957,7 @@ impl<'a, 'db> ArgumentTypeChecker<'a, 'db> {
         signature: &'a Signature<'db>,
         arguments: &'a CallArguments<'a>,
         argument_types: &'a [Type<'db>],
-        argument_parameters: &'a [Option<usize>],
+        argument_parameters: &'a [SmallVec<[usize; 1]>],
```

Comment:
nit: At this point I think we're using this pattern in multiple places. Do you think it might be useful to use something like https://docs.rs/vec1/1.12.1/vec1/smallvec_v1/index.html ?